src/vs/workbench/api/common/extHost.protocol.ts

```diff
@@ -2385,12 +2385,13 @@ export interface ExtHostTerminalServiceShape {
 	$provideTerminalQuickFixes(id: string, matchResult: TerminalCommandMatchResultDto, token: CancellationToken): Promise<SingleOrMany<TerminalQuickFix> | undefined>;
 }
 
-export interface ExtHostTerminalShellIntegrationShape {
+export interface ExtHostTerminalShellIntegrationShape { // what is this shape for?
 	$shellIntegrationChange(instanceId: number): void;
 	$shellExecutionStart(instanceId: number, commandLineValue: string, commandLineConfidence: TerminalShellExecutionCommandLineConfidence, isTrusted: boolean, cwd: UriComponents | undefined): void;
 	$shellExecutionEnd(instanceId: number, commandLineValue: string, commandLineConfidence: TerminalShellExecutionCommandLineConfidence, isTrusted: boolean, exitCode: number | undefined): void;
 	$shellExecutionData(instanceId: number, data: string): void;
-	$cwdChange(instanceId: number, cwd: UriComponents | undefined): void;
+	$shellEnvChange(); // what should be the parameters/arguments?
```

Comment:
Probably can't use map, may have to be array of map entry. 
Search file for similar thing